### DISCUSSION TITLE
Add Android to Compatibility matrix

### DISCUSSION THREAD

**Discussion by marcprux** [SwiftPackageIndex/SwiftPackageIndex-Server#3461] [Ideas]
An installable Swift 6.0.1 SDK Bundle for Android has recently been published at https://github.com/finagolfin/swift-android-sdk. Android would make a great addition to the build compatibility list. And unlike the [windows support issue](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/discussions/1354), this should be fairly straightforward to implement, since it can be added onto the Linux build. After running `swift sdk install …`, you'd simply need to compile with `swift build --build-tests --swift-sdk aarch64-unknown-linux-android24`.

**Comment by finestructure** [comment]
That's a very interesting development, thanks for sharing! Are there any plans lands these SDKs somewhere under `swiftlang` at some point?

**Comment by marcprux** [comment]
Yes, we will eventually be adding the Android SDK to the Swift community CI system with the hopes of graduating to a fully-supported platform. That's a long process (Windows is still not there after many years), but offering people insights into which projects are Android-compatible will be a good first step along this road.

**Comment by marcprux** [comment]
I've set up a supplementary site at https://swift-everywhere.org/ that builds all the Linux-compatible packages for Android. 1,926 out of 3,321 packages (57%) are currently building successfully, which isn't a bad start. More will probably start passing once support for the [Testing module](https://github.com/finagolfin/swift-android-sdk/issues/174) is in place.

If anyone is interested in supporting Android in their packages, check that list to see if your repository is listed, and consider adding the [swift-android-action](https://github.com/marketplace/actions/swift-android-action) to your CI workflows.

**Comment by marcprux** [comment]
Android build results are now being published: https://swiftpackageindex.com/blog/adding-wasm-and-android-compatibility-testing. Hooray!
